### ISSUE TITLE
popover: replace current popover with the popover-next

### ISSUE THREAD

**Issue by spike-rabbit** [siemens/element#389]
We need to replace the old popover with the popover next,  as the old popover lacks a11y features.
This task needs to be split in multiple steps:

1. fix the current popover API by prefixing all input and outputs with `siPopoverNext`
2. add @internal where needed
3. rename the old popover to popover-legacy (with a breaking commit)
4. rename popover-next to popover.

**Comment by kfenner** [comment]
@mistrykaran91 I guess this is all done now, right?
Closing...
